### PR TITLE
Adds support to publish as scala.js library

### PR DESCRIPTION
**Pull request by Voltir**
Im not entirely sure if this works with the publishSettings value you have to publish to sonatype, but `publishLocal` works.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -17,6 +19,9 @@ lazy val core = project.in(file("core"))
     tutTargetDirectory := file(".")
   )
 
+lazy val coreJS = core.js
+lazy val coreJVM = core.jvm
+
 lazy val commonSettings = Def.settings(
```
Comment (DavidGregory084) [inline build.sbt]: I don't really know a lot about how the Scala.js plugin works - do the `libraryDependencies` in the `commonSettings` need to use the `%%%` separator?

**Review by DavidGregory084 — COMMENTED** [review]
Looks perfect to me, I just have one question about the libraryDependencies. Thanks @Voltir!

File: build.sbt (hunk 1 above)
Comment (Voltir) [reply build.sbt]: You know, im not entirely sure. As far as I can tell, `%%%` wasn't needed, but maybe there is a runtime linker problem waiting to blow up if I use it more.

Comment (DavidGregory084) [reply build.sbt]: Ok, let's merge it 👍

**Pull request merged**